jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesTest.java

```diff
@@ -347,6 +347,7 @@ public void testHumanTaskProcessBeforeAndAfterUndeploy() {
 
         // now let's undeploy the unit
         deploymentService.undeploy(deploymentUnit);
+        units.remove(deploymentUnit);
```

Comment:
why this ? the change to RuntimeDataServiceImpl.java looks to me like completely compatible regarding this.

Reply:
This change was suggested by Gonzalo. The target is to keep client units array consistent with the server.

Reply:
Yes, that's the reason, it's the same in other tests:
https://github.com/kiegroup/jbpm/blob/74b525d1454e80de85ae02a8a5ffe77bf85706e7/jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/archive/EJBService.java#L261-L263